### PR TITLE
replace xarray merge with combine_first, deal with dates

### PR DESCRIPTION
**Pull request by cgaustin**

### CODE REVIEW

File: cw/worker.py [hunk 1]
```diff
@@ -119,12 +121,14 @@ def rainbow(self, x, y, t, specs_url, tiles_url, requested_ubids):
...
                     if band:
-                        ds = ds.merge(band)
+                        # combine_first instead of merge, for locations where data is missing for some bands
+                        ds = ds.combine_first(band)
         return ds
 
     def detect(self, rainbow, x, y):
         """ Return results of ccd.detect for a given stack of data at a particular x and y """
         try:
+            rainbow_date_array = np.array(rainbow['t'].values)
```
Comment (jmorton) [inline cw/worker.py]: Are you sure we can't calculate this once and set it as `coords` on the array?

Comment (cgaustin) [reply cw/worker.py]: i see. like on line 126 instead

File: cw/worker.py [hunk 2]
```diff
@@ -108,6 +109,7 @@ def landsat_dataset(self, spectrum, x, y, t, ubid, specs_url, tiles_url):
         ds.coords['t']        = (('t'), pd.to_datetime([t['acquired'] for t in tiles]))
         ds.coords['source']   = (('t'), [t['source'] for t in tiles])
         ds.coords['acquired'] = (('t'), [t['acquired'] for t in tiles])
+        # the following can probably go away, since it gets wiped out during xarray combine_first()
```
Comment (jmorton) [inline cw/worker.py]: It seems strange that this would get wiped out but source and acquired would not...

Comment (cgaustin) [reply cw/worker.py]: they might actually, they werent/aren't referenced in the detect function

**Pull request merged**